### DISCUSSION TITLE
no matches for kind "PodMonitor" in version "monitoring.coreos.com/v1

### DISCUSSION THREAD

**Discussion by vinycoolguy2015** [fluxcd/flux2#3189] [Q&A] [answered]
tried installing the monitorin stack with flux in GKE and got this error

no matches for kind "PodMonitor" in version "monitoring.coreos.com/v1

gke version v1.22.12-gke.1200

**Comment by stefanprodan** [comment, marked as answer]
Have you deployed the kube-prometheus-stack HelmRelease? See https://fluxcd.io/flux/guides/monitoring/#install-the-prometheus-stack

**Reply by vinycoolguy2015** [reply]
Thanks.managed to get it fixed

**Reply by newtondotcom** [reply]
How ?
I am trying to install [flux2-monitoring-example](https://github.com/fluxcd/flux2-monitoring-example) and i am facing this error, do i need to install `kube-prometheus-stack HelmRelease` before ?

**Reply by matheuscscp** [reply]
Yes

**Reply by newtondotcom** [reply]
Thanks !
Apparently, the issue i had was that in the [conf file](https://github.com/fluxcd/flux2-monitoring-example/blob/main/clusters/test/monitoring.yaml), the `spec.dependsOn` condition was not respected.
So I temp removed the `configs` folder and `Kustomization` , run, push and then bring back `configs`.
It might be an issue in my specific setup or not, I dont have time to reproduce it and maybe open an issue ! In case this is not a bug, might be good to specify it in the docs !
